api/test/e2e/consumer_test.go

```diff
@@ -0,0 +1,172 @@
+ * limitations under the License.
+ */
+package e2e
+
+import (
+	"net/http"
+	"testing"
+)
+
+//CASE 1: add consumer with username
+func TestConsumer_add_consumer_with_username(t *testing.T) {
+	tests := []HttpTestCase{
+		{
+			caseDesc: "create consumer",
+			Object:   MangerApiExpect(t),
+			Path:     "/apisix/admin/consumers",
+			Method:   http.MethodPut,
+			Body: `{
+				 "username": "jack",
+				 "plugins": {
+					 "key-auth": {
+						 "key": "auth-one"
+					 }
+				 },
+				 "desc": "test description"
+			 }`,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+		},
+		{
+			caseDesc: "create route",
+			Object:   MangerApiExpect(t),
+			Method:   http.MethodPut,
+			Path:     "/apisix/admin/routes/r1",
+			Body: `{
+				 "uri": "/hello",
+				 "plugins": {
+					 "key-auth": {}
+				 },
+				 "upstream": {
+					 "type": "roundrobin",
+					 "nodes": {
+						 "172.16.238.20:1980": 1
+					 }
+				 }
+			 }`,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+		},
+		{
+			caseDesc:     "verify route",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			Headers:      map[string]string{"apikey": "auth-one"},
+			ExpectStatus: http.StatusOK,
+			Sleep:        sleepTime, //sleep x millisecond before verify route
+		},
+		{
+			caseDesc:     "verify route without auth",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			ExpectStatus: http.StatusUnauthorized,
+			Sleep:        sleepTime,
+		},
+	}
+
+	for _, tc := range tests {
+		testCaseCheck(tc)
+	}
+}
+
+//CASE 2: add consumer without username
+func TestConsumer_add_consumer_without_username(t *testing.T) {
+	tests := []HttpTestCase{
+		{
+			caseDesc: "create consumer",
+			Object:   MangerApiExpect(t),
+			Path:     "/apisix/admin/consumers",
+			Method:   http.MethodPut,
+			Body: `{
+				 "plugins": {
+					 "key-auth": {
+						 "key": "auth-new"
+					 }
+				 },
+				 "desc": "test description"
+			 }`,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusBadRequest,
+		},
+		{
+			caseDesc:     "verify route",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			Headers:      map[string]string{"apikey": "auth-new"},
+			ExpectStatus: http.StatusUnauthorized,
+			Sleep:        sleepTime,
+		},
+	}
+
+	for _, tc := range tests {
+		testCaseCheck(tc)
+	}
+}
+
+//CASE 3: add consumer with labels
+
+//CASE 4: delete consumer
+func TestConsumer_delete_consumer(t *testing.T) {
+	tests := []HttpTestCase{
+		{
+			caseDesc:     "delete consumer",
+			Object:       MangerApiExpect(t),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/consumers/jack",
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+		},
```

Comment:
need a test case for deleting an un-exist consumer.

Reply:
ok, thanks for the suggestion.